### PR TITLE
Add support for private keys provided as strings

### PR DESCRIPTION
**Pull request by jasper-lyons**
> Note: happy to update documentation for this once merged.

This request provided the ability for private keys to be provided as strings.

**Use Case**: When working with systems with ephemeral filesystems one would need to put the key file on the server, potentially committing it to git which is a fairly large [security hole](https://help.github.com/articles/remove-sensitive-data/).

This means that you can safely use this plugin on Heroku.

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -1,31 +1,16 @@
...
-# Load fixtures from the engine
-if ActiveSupport::TestCase.respond_to?(:fixture_path=)
-  ActiveSupport::TestCase.fixture_path = File.expand_path('../fixtures', __FILE__)
-  ActionDispatch::IntegrationTest.fixture_path = ActiveSupport::TestCase.fixture_path
-  ActiveSupport::TestCase.fixtures :all
-end
-
 require 'webmock/minitest'
+
+#Minitest::Reporters.use!
```
Comment (lgromanowski) [inline test/test_helper.rb]: Why did you removed minitest reporters (and also minitest backtrace filter)?

File: .byebug_history [hunk 2]
```diff
@@ -0,0 +1,22 @@
+continue
```
Comment (lgromanowski) [inline .byebug_history]: Is this file needed?

**Comment by lgromanowski** [comment]
Unfortunately I can't merge it at this moment - 3 test are failing:
- test_if_challenge_is_valid
- test_if_challenge_request_is_invalid_if_it_is_larger_than_256_bytes
- test_if_challenge_request_is_invalid_when_is_smaller_than_128_bits

File: test/test_helper.rb (hunk 1 above)
Comment (jasper-lyons) [inline test/test_helper.rb]: I just found it difficult to parse while writing tests so changed it back. If it's required to have then I can add them back. There is a number of other things I should probably clean up

* byebug_history etc.

Comment (lgromanowski) [inline test/test_helper.rb]: Yes, please restore minitest reporters

**Comment by jasper-lyons** [comment]
Figured out what my issue was.

The fixtures persisted in the db after my initial test runs and so when removing code form the test_helper and running test to see if things worked there was a hidden bug, the fixtures were not loaded.

Might be worth using a db cleaner? Either way, latest push should fix the things I broke!

**Comment by lgromanowski** [comment]
Merged, thanks! :)
